src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotPrimaryDataQueryInterface.java

```diff
@@ -21,17 +20,15 @@ default StepResult prepareQueryAndCreateSnapshot(
       FlightContext context,
       Snapshot snapshot,
       SnapshotRequestModel snapshotReq,
-      DatasetService datasetService)
+      Dataset sourceDataset)
       throws InterruptedException {
```

Comment:
can you remind me why the code for the byQuery mode needs to change? Are we still using this code for the byRequestId mode?

Reply:
Its general cleanup. We want to use Dataset where we can, since it is cleaner than refetching it all over the place.